blob/azureblob/azureblob.go

```diff
@@ -481,6 +481,23 @@ func openBucket(ctx context.Context, pipeline pipeline.Pipeline, accountName Acc
 		blobURL.RawQuery = strings.TrimPrefix(string(opts.SASToken), "?")
 	}
 	serviceURL := azblob.NewServiceURL(*blobURL, pipeline)
+
+	if opts.Credential == nil {
```

Comment:
Could you get credentials earlier using https://godoc.org/github.com/Azure/azure-storage-blob-go/azblob#NewUserDelegationCredential ?

Alternatively, could you do this lazily in SignURL?

Either way, please update the docstring for Options.Credential.